request-manager/graph/grapher.go

```diff
@@ -217,25 +236,27 @@ func buildSeqGraph(seqSpec *spec.Sequence, idgen id.Generator) (seqGraph *Graph,
 		}
 	}
 
-	// Nodes we've yet to add
+	// Nodes we've yet to add, represented as single-node graphs
 	nodesToAdd := map[string]*Graph{}
 	for k, v := range nodes {
 		nodesToAdd[k] = v
 	}
 	nodesAdded := map[string]bool{}
 
 	// Build graph by adding nodes, starting from the source node, and then
-	// adding all adjacent nodes to the source node, and so on.
-	// We cannot add nodes in any order because we do not know the reverse dependencies
+	// adding all adjacent nodes to the source node, and so on
+	// We cannot add nodes in any order because we do not know the reverse
+	// dependencies
 	for len(nodesToAdd) > 0 {
 
-		componentAdded := false
+		nodeAdded := false
 
-		for nodeName, component := range nodesToAdd {
+		for nodeName, node := range nodesToAdd {
 			nodeSpec := seqSpec.Nodes[nodeName]
 			if dependenciesSatisfied(nodesAdded, nodeSpec.Dependencies) {
```

Comment:
```go
if !dependenciesSatisfied(nodesAdded, nodeSpec.Dependencies) {
    continue
}

// rest of the code
```

From https://golang.org/doc/effective_go.html#if:

> This is an example of a common situation where code must guard against a sequence of error conditions. The code reads well if the successful flow of control runs down the page, eliminating error cases as they arise. 

Go code should "read down the left".